### PR TITLE
Promises

### PR DESCRIPTION
**Pull request by houli**

### CODE REVIEW

File: www/js/serv-storage.js [hunk 1]
```diff
@@ -69,23 +70,22 @@ angular.module('starter.services', [])
...
 
             sending = [];
             localStorage.setItem(sendingMemAdd, JSON.stringify(sending));
-            return true;
+            deferred.resolve(200);
           }
           else {
             dirty = JSON.parse(localStorage.getItem(dirtyMemAdd)) || dirty;
-            for (var i = response.results.error_checksums.length - 1; i >= 0; i--) {
+            for (var i = response.data.results.error_checksums.length - 1; i >= 0; i--) {
```
Comment (houli) [inline www/js/serv-storage.js]: IDK what's going on here but surely using nice functions like forEach, filter map etc. will make this a lot easier to read

**Comment by houli** [comment]
Some of the most promising code I've seen in my time.

**Comment by houli** [comment]
Don't think I need this anymore 😆

**Comment by 22a** [comment]
Distreeb

**Comment by 22a** [comment]
Speak up, I don't have my ear trumpet

**Pull request closed without merging**